src/custom-element.js

```diff
@@ -123,48 +123,55 @@ export function stubElements(win) {
  * @param {!AmpElement} element
  */
 export function applyLayout_(element) {
+  const layoutAttr = element.getAttribute('layout');
   let widthAttr = element.getAttribute('width');
   let heightAttr = element.getAttribute('height');
   const sizesAttr = element.getAttribute('sizes');
-  const layoutAttr = element.getAttribute('layout');
+  let layout = null;
+  let width = null;
+  let height = null;
+
+  // Layout specified directly.
+  if (layoutAttr) {
+    layout = parseLayout(layoutAttr.trim());
```

Comment:
Quite sure the trim() can go, since there's no currently considered valid document that would need it.